src/func.cpp

```diff
@@ -993,6 +991,29 @@ void FunctionTemplate::Print() const {
     fflush(stdout);
 };
 
+llvm::GlobalValue::LinkageTypes lGetTemplateInstantiationLinkage(TemplateInstantiationKind kind) {
+    llvm::GlobalValue::LinkageTypes linkage = llvm::GlobalValue::LinkageTypes::InternalLinkage;
+    switch (kind) {
+    // Function can be defined multiple times across different translation units without causing conflicts.
+    // The linker will choose a definition for the function based on its default behavior.
+    case TemplateInstantiationKind::Explicit:
+        linkage = llvm::GlobalValue::WeakODRLinkage;
+        break;
+    // The function is only allowed to be defined once across all translation units, but it can be discarded if unused.
+    // If multiple definitions of the function are present across different translation units, the             linker
```

Comment:
to many whitespaces in the end of the line :)

Reply:
Fixed